### PR TITLE
DOCS-627 Error documentation

### PR DESCRIPTION
**Pull request by rachelnabors**
This PR seeks to provide documentation for Clerk's errors ([source available here](https://github.com/clerk/clerk_go/tree/main/api/apierror)).

Its format is inspired by [Plaid's error docs](https://plaid.com/docs/errors/).

It is incomplete and would need help from Support to fill out. 

## Examples

* [Error overview page](https://docs-preview-580.clerkpreview.com/docs/errors/overview)
* [Actor token errors page](https://docs-preview-580.clerkpreview.com/docs/errors/actor-tokens)

## Next steps

[] Work with Support to round out the Actor tokens page
[] After settling on a format for these pages, document the rest of the errors

### CODE REVIEW

**Comment by royanger** [comment]
I would suggest adding these errors and shipping this PR. We can add the rest later.

https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/sign_in.go

https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/sign_in_tokens.go

https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/sign_up.go

File: docs/errors/auth.mdx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+## `InvalidHandshake`
+
+```json
+{
+    shortMessage: "invalid handshake",
+    longMessage:  "The handshake request is invalid: <REASON>",
+    code:         invalid_handshake_code,
+}
```
Comment (alexisintech) [inline docs/errors/auth.mdx:260]: missing closing backticks

File: docs/errors/blocklist-identifiers.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+## `BlocklistIdentifierNotFound`
+
+```json
+{
+    shortMessage: "Identifier not found",
+    longMessage:  "No identifier was found with id <identifierID>",
+    code:         resource_not_found_code,
+}
```
Comment (alexisintech) [inline docs/errors/blocklist-identifiers.mdx:26]: missing closing backticks

File: docs/errors/clients.mdx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+## `ClientNotFoundInRequest`
+
+Signifies an error when no client is found in an incoming request.
+
+```json
+{
+    shortMessage: "No client found",
+    longMessage:  "This request is expecting a client and did not find one",
+    code:         client_not_found_code,
+}
```
Comment (alexisintech) [inline docs/errors/clients.mdx:30]: missing closing backticks

File: docs/errors/domains.mdx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+```
+
+## `InvalidProxyConfiguration`
+
+```json
+{
+    longMessage:  "Clerk Frontend API cannot be accessed through the proxy URL. Make sure your proxy is configured correctly.",
+    code:         invalid_proxy_configuration_code,
+}
```
Comment (alexisintech) [inline docs/errors/domains.mdx:85]: missing closing backticks

**Comment by JacobMGEvans** [comment]
Im picking this up this week.

File: docs/errors/auth.mdx (hunk 1 above)
Comment (JacobMGEvans) [reply docs/errors/auth.mdx:260]: Did these get resolved Alexis?

Comment (JacobMGEvans) [reply docs/errors/auth.mdx]: NVM added them

File: docs/errors/domains.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+{
+    "shortMessage": "Domain not found",
+    "longMessage":  "No domain was found with <id>",
+    "code":         "resource_not_found_code"
+}
+```
+
+## `DomainUpdateForbidden`
+
+Signifies an error when trying to update an non production instance domain.
```
Comment (S3Prototype) [inline docs/errors/domains.mdx:23]:
```suggestion
Signifies an error when trying to update a non production instance domain.
```

File: docs/errors/domains.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+{
+    "shortMessage": "sync nonce already consumed",
+    "longMessage":  "The given sync nonce has already been consumed and cannot be re-used.",
+    "code":         "sync_nonce_already_consumed_code"
+}
+```
+
+## `PrimaryDomainAlreadyExists`
+
+Signifies an error when a new domain is added as primary when there is already once in the instance. Currently, we only support a single primary domain per instance.
```
Comment (S3Prototype) [inline docs/errors/domains.mdx:68]:
```suggestion
Signifies an error when a new domain is added as primary while there is already one in the instance. Currently, we only support a single primary domain per instance.
```

File: docs/errors/applications.mdx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+title: Applications errors
+description: An index of Clerk's applications errors.
+type: reference
+---
+
+# Application errors index
+
+## `ApplicationNotFound`
+
+Signifies an error when no application with given appID was found.
```
Comment (S3Prototype) [inline docs/errors/applications.mdx:11]:
```suggestion
Signifies an error when no application with the given appID was found.
```

**Review by S3Prototype — APPROVED** [review]
Approved with suggested changes

File: docs/errors/domains.mdx (hunk 5 above)
Comment (alexisintech) [reply docs/errors/domains.mdx:23]:
```suggestion
Signifies an error when trying to update a domain for any instance that is not a production instance.
```

File: docs/errors/applications.mdx (hunk 7 above)
Comment (alexisintech) [reply docs/errors/applications.mdx:11]:
```suggestion
Signifies an error when no application with the given `appID` was found.
```

File: docs/errors/db.mdx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+## `DuplicateAllowlistIdentifier`
+
+```json
+{
+    "shortMessage": "duplicate allowlist identifier",
+    "longMessage":  "the identifier <identifier> already exists",
+    "code":         "duplicate_record_code"
+}
+```
```
Comment (alexisintech) [inline docs/errors/db.mdx:19]: I doubt this is the correct error meant for this page. This needs to get moved to the allowlist identifiers page, no?
Or does DB stand for something else, and I was mistaken to change it to "Database"?

File: docs/errors/comms.mdx [hunk 9]
```diff
@@ -0,0 +1,63 @@
+---
+title: Comms errors
+description: An index of Clerk errors related to comms.
```
Comment (alexisintech) [inline docs/errors/comms.mdx:3]: what is comms?

I see these errors are about Identification. Can we come up with a more specific title for this page?

**Review by alexisintech — CHANGES_REQUESTED** [review]

File: docs/errors/comms.mdx (hunk 9 above)
Comment (JacobMGEvans) [reply docs/errors/comms.mdx:3]: 🤷🏻 https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/comms.go

File: docs/errors/db.mdx (hunk 8 above)
Comment (alexisintech) [reply docs/errors/db.mdx:19]: was there anyone you were working with on team sdk to get these docs out ? if not, would you mind reaching out to them to get this sorted ?

**Comment by alexisintech** [comment]
I would like the DuplicateAllowlistIdentifier moved out of the ["db" errors](https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/db.go) and into the ["allowlist identifiers" errors](https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/allowlist_identifiers.go)
I would also like the "comms" errors to get consolidated into the ["identifications" errors](https://github.com/clerk/clerk_go/blob/3df6867bfb2913c1ab95131a2bf49dee29d4c0c3/api/apierror/identifications.go)

currently waiting on the backend engineers to reply https://clerkinc.slack.com/archives/C05GH6N5LLS/p1708471100599579

File: docs/errors/db.mdx (hunk 8 above)
Comment (royanger) [reply docs/errors/db.mdx:19]: Some of these errors will never get sent out to the frontend or backend. This is probably one of them, and is the correct error.

File: docs/errors/overview.mdx
Comment (alexisintech) [inline docs/errors/overview.mdx:1]: This is good to merge; the only qualm I have is that this errors overview page is heinous to maintain. imo, I think the headings can link to the respective pages, but I do not think that every single page listed should have a link.

also, the overview isn't finished - it's missing "sign-in", "sign-up", and "sign-in-tokens"

**Review by alexisintech — APPROVED** [review]

**Pull request merged**